packages/web/components/navbar-osmo-price.tsx

```diff
@@ -0,0 +1,196 @@
+    chainStore.osmosis.stakeCurrency.coinDenom
+  );
+
+  if (!osmoPrice || !osmoCurrency) return null;
+
+  return (
+    <div className="flex flex-col gap-6 px-2">
+      <div className="flex items-center justify-between  px-2">
+        <SkeletonLoader isLoaded={osmoPrice.isReady} className="min-w-[70px]">
+          <div className="flex items-center gap-1">
+            <div className="h-[20px] w-[20px]">
+              <Image
+                src={osmoCurrency.coinImageUrl!}
+                alt="Osmo icon"
+                width={20}
+                height={20}
+              />
+            </div>
+
+            <p className="mt-[3px]">
+              {osmoPrice.fiatCurrency.symbol}
+              {Number(osmoPrice.toDec().toString()).toFixed(2)}
+            </p>
+          </div>
+        </SkeletonLoader>
+
+        <SkeletonLoader
+          isLoaded={
+            !tokenDataQuery.isFetching &&
+            !tokenChartQuery.isFetching &&
+            osmoPrice.isReady
+          }
+          className="flex min-h-[23px] min-w-[85px] items-center justify-end gap-1.5"
+        >
+          <Sparkline
+            data={getChartData(tokenChartQuery?.getChartPrices)}
+            width={25}
+            height={24}
+            lineWidth={2}
+          />
+
+          <p
+            className={
+              tokenDataQuery.get24hrChange?.toDec().gte(new Dec(0))
+                ? "text-bullish-400"
+                : "text-error"
+            }
+          >
+            {tokenDataQuery.get24hrChange
+              ?.maxDecimals(2)
+              .inequalitySymbol(false)
+              .toString()}
+          </p>
+        </SkeletonLoader>
+      </div>
+
+      {account.walletStatus === WalletStatus.Loaded && (
+        <SkeletonLoader isLoaded={osmoPrice.isReady}>
+          <Button
+            mode="tertiary"
+            className={classNames(
+              "button group relative flex !h-11 items-center justify-center gap-2 overflow-hidden !rounded-full !border-osmoverse-700 !py-1 font-bold text-osmoverse-100 !transition-all !duration-300 !ease-in-out",
+              "hover:border-none hover:bg-gradient-positive hover:text-osmoverse-1000"
+            )}
+            onClick={() => transferConfig.buyOsmo()}
+          >
+            <CreditCardIcon />{" "}
+            <span className="z-10 mt-0.5 flex-shrink-0">
+              {t("menu.buyTokens")}
+            </span>{" "}
+            <CoinsIcon
+              className={classNames(
+                "invisible absolute top-0 -translate-y-full transform transition-transform ease-linear",
+                "group-hover:visible group-hover:translate-y-[30%] group-hover:duration-[3s]"
+              )}
+            />
+          </Button>
+        </SkeletonLoader>
+      )}
+
+      {transferConfig?.fiatRampsModal && (
+        <FiatRampsModal {...transferConfig.fiatRampsModal} />
+      )}
+    </div>
+  );
+});
+
+const CreditCardIcon = () => {
```

Comment:
Let's move this into the components/assets folder

Reply:
I was thinking about it, but the component has some deep CSS classes in the `path`s that only apply to this use case.